### ISSUE TITLE
[BUG] Eclipse cannot open Java files after installing Lombok

### ISSUE THREAD

**Issue by yannickkirschen** [projectlombok/lombok#2825]
**Describe the bug**

When I install Lombok for Eclipse (via [this](https://projectlombok.org/setup/eclipse) instruction), Eclipse cannot open Java source files anymore and crashes with the message: `Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @39e43310`.

I am quite sure that this goes along with Lombok. When I make a fresh installation of Eclipse, everything works fine. Only after I installed Lombok, it crashes.

I found another instruction ([here](https://www.baeldung.com/lombok-ide#eclipse)) that says I should be able to see a note in the about screen of Eclipse that Lombok is installed. That note is not there.

**To Reproduce**

- Install Eclipse
- Download the Lombok Jar file
- `java -jar <lombok.jar>`
- Install Lombok for Eclipse
- Quit the installer
- Restart Eclipse if already open; else start Eclipse
- Open a Java file of your choice

**Expected behavior**

When you open a Java file, Eclipse will show a message like this in the view that should contain the source code: `Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @39e43310`.

**Version info:**

- Lombok: 1.18.20
- Eclipse: 2021-03 (4.19.0) (Build: 20210312-0638)
- `javac -version`: javac 16
- `java --version`:
    - openjdk 16 2021-03-16
    - OpenJDK Runtime Environment (build 16+36-2231)
    - OpenJDK 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)
- macOS 11.3

**Comment by Rawi01** [comment]
This is a duplicate of #2810

As a workaround you can use a Java version < 16, build the current code yourself, use [this pre-built version](https://github.com/projectlombok/lombok/files/6369283/lombok.zip) or add `--illegal-access=warn --add-opens java.base/java.lang=ALL-UNNAMED` to your `eclipse.ini`.
